packages/flutter_tools/lib/src/isolated/native_assets/targets.dart

```diff
@@ -0,0 +1,378 @@
+            .map(getNativeAndroidArchitecture)
+            .map(
+              (Architecture architecture) => AndroidAssetTarget(
+                architecture: architecture,
+                supportedAssetTypes: supportedAssetTypes,
+                environmentDefines: environmentDefines,
+              ),
+            )
+            .toList();
+      case TargetPlatform.ios:
+        final List<DarwinArch> iosArchs =
+            _emptyToNull(
+              environmentDefines[kIosArchs],
+            )?.split(' ').map(getIOSArchForName).toList() ??
+            <DarwinArch>[DarwinArch.arm64];
+        return iosArchs
+            .map(getNativeIOSArchitecture)
+            .map(
+              (Architecture architecture) => IOSAssetTarget(
+                environmentDefines: environmentDefines,
+                fileSystem: fileSystem,
+                architecture: architecture,
+                supportedAssetTypes: supportedAssetTypes,
+              ),
+            )
+            .toList();
+      case TargetPlatform.web_javascript:
+        return <AssetBuildTarget>[WebAssetTarget(supportedAssetTypes: supportedAssetTypes)];
+      case TargetPlatform.tester:
+        return <AssetBuildTarget>[
+          FlutterTesterAssetTarget(supportedAssetTypes: supportedAssetTypes),
+        ];
+      case TargetPlatform.fuchsia_arm64:
+      case TargetPlatform.fuchsia_x64:
+        throw UnsupportedError('No targets defined for target platform $targetPlatform.');
+    }
+  }
+}
+
+final class WebAssetTarget extends AssetBuildTarget {
+  WebAssetTarget({required super.supportedAssetTypes});
+
+  @override
+  List<ProtocolExtension> get extensions => <ProtocolExtension>[...dataAssetExtensions];
+
+  @override
+  String get targetString => 'web';
+}
+
+sealed class CodeAssetTarget extends AssetBuildTarget {
+  CodeAssetTarget({
+    required super.supportedAssetTypes,
+    required this.architecture,
+    required this.os,
+  });
+
+  final Architecture architecture;
+  final OS os;
+
+  late final CCompilerConfig? cCompilerConfigSync;
+
+  Future<void> setCCompilerConfig();
+
+  List<CodeAssetExtension> get codeAssetExtensions => <CodeAssetExtension>[
+    if (supportedAssetTypes.contains(SupportedAssetTypes.codeAssets))
+      CodeAssetExtension(
+        targetArchitecture: architecture,
+        linkModePreference: LinkModePreference.dynamic,
+        cCompiler: cCompilerConfigSync,
+        targetOS: os,
+      ),
+  ];
+
+  @override
+  String get targetString => '${os.name}_${architecture.name}';
+}
+
+class WindowsAssetTarget extends CodeAssetTarget {
+  WindowsAssetTarget({required super.supportedAssetTypes, required super.architecture})
+    : super(os: OS.windows);
+
+  @override
+  List<ProtocolExtension> get extensions => <ProtocolExtension>[
+    ...codeAssetExtensions,
+    ...dataAssetExtensions,
+  ];
+
+  @override
+  Future<void> setCCompilerConfig() async => cCompilerConfigSync = await cCompilerConfigWindows();
```

Comment:
Nit: I'd use curly braces here and the other implementations of this function within this file for readability.